src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java

```diff
@@ -147,6 +155,46 @@ public void testRunRequestAsync() throws Exception {
     }
 
     @Test
+    public void testRunRequestWithSocketTimeout() throws Exception {
+        exception.expect(SaltStackException.class);
+        exception.expectMessage(containsString("Read timed out"));
+
+        // create a local SaltStackClient with a fast timeout configuration 
+        // to do not lock tests more thant 2s
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        SaltStackClient clientWithFastTimeout = new SaltStackClient(uri);
+        clientWithFastTimeout.getConfig().put(SOCKET_TIMEOUT, 1000);
```

Comment:
Because you do not test a default timeout? :-) In this case you can put 50 ms as well, as a positive non-zero will set a timeout as per se.